code/modules/projectiles/guns/powered.dm

```diff
@@ -0,0 +1,123 @@
+/obj/item/gun/ballistic/automatic/powered
+	mag_type = /obj/item/ammo_box/magazine/gauss
+	can_suppress = FALSE
+
+	var/obj/item/stock_parts/cell/cell
+	var/cell_type = /obj/item/stock_parts/cell/gun
```

Comment:
you can merge these into one variable maybe? start as a typepath and turn it into an object in init

Reply:
Doing this did some very strange effects, where it didnt take the charge from the gun cell and fired without it?

Reply:
how were you doing it?

Reply:
at first, i was doing it like:

cell = new cell(src)

then,

cell = new(src)

neither of which worked